From d3ae7e70ed16bc3e4dc6078fa633dc0c6fb73732 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?David=20D=C3=ADaz-Guerra=20Aparicio?= <ddga@unizar.es>
Date: Mon, 22 May 2023 15:18:29 +0300
Subject: [PATCH 1/3] Fix pandas append deprecation

DataFrame.append and Series.append were flagged as deprecated in pandas 1.4.0 and finally removed in pandas 2.0.0. I've just replaced them by pandas.concat as suggested in https://pandas.pydata.org/docs/whatsnew/v1.4.0.html#whatsnew-140-deprecations-frame-series-append
---
 museval/aggregate.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/museval/aggregate.py b/museval/aggregate.py
index bea39a3..6c24a98 100644
--- a/museval/aggregate.py
+++ b/museval/aggregate.py
@@ -180,9 +180,9 @@ def add_track(self, track):
             track store object
         """
         if isinstance(track, TrackStore):
-            self.df = self.df.append(track.df, ignore_index=True)
+            self.df = pd.concat([self.df, track.df], ignore_index=True)
         else:
-            self.df = self.df.append(track, ignore_index=True)
+            self.df = pd.concat([self.df, track], ignore_index=True)
 
     def add_eval_dir(self, path):
         """add precomputed json folder to dataframe
@@ -299,7 +299,7 @@ def add_sisec18(self):
         raw_data = urlopen('https://github.com/sigsep/sigsep-mus-2018-analysis/releases/download/v1.0.0/sisec18_mus.pandas')
         print('Done!')
         df_sisec = pd.read_pickle(raw_data, compression=None)
-        self.df = self.df.append(df_sisec, ignore_index=True)
+        self.df = pd.concat([self.df, df_sisec], ignore_index=True)
 
     def add_eval_dir(self, path):
         """add precomputed json folder to dataframe.
@@ -336,7 +336,7 @@ def add_evalstore(self, method, name):
         """
         df_to_add = method.df
         df_to_add['method'] = name
-        self.df = self.df.append(df_to_add, ignore_index=True)
+        self.df = pd.concat([self.df, df_to_add], ignore_index=True)
     
     def agg_frames_scores(self):
         """aggregates frames scores

From 0451a1e378800cef624986d34857cce60cecd372 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?David=20D=C3=ADaz-Guerra=20Aparicio?= <ddga@unizar.es>
Date: Mon, 22 May 2023 15:28:33 +0300
Subject: [PATCH 2/3] Fix new default behaviour of simplejson

Since version  v3.19.0 simplejson.load doesn't allow NaN values, so the code crashes. I've just added allow_nan=True when calling the function, but this is not available in order versions of simplejson.
---
 museval/aggregate.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/museval/aggregate.py b/museval/aggregate.py
index 6c24a98..4285542 100644
--- a/museval/aggregate.py
+++ b/museval/aggregate.py
@@ -110,7 +110,7 @@ def df(self):
             pandas dataframe object of track scores
         """
         # encode and decode to json first
-        return json2df(simplejson.loads(self.json), self.track_name)
+        return json2df(simplejson.loads(self.json, allow_nan=True), self.track_name)
 
     def __repr__(self):
         """Print the frames_aggregated values instead of all frames
@@ -197,7 +197,7 @@ def add_eval_dir(self, path):
             json_paths = p.glob('test/**/*.json')
             for json_path in json_paths:
                 with open(json_path) as json_file:
-                    json_string = simplejson.loads(json_file.read())
+                    json_string = simplejson.loads(json_file.read(), allow_nan=True)
                 track_df = json2df(json_string, json_path.stem)
                 self.add_track(track_df)
 
@@ -317,7 +317,7 @@ def add_eval_dir(self, path):
             json_paths = p.glob('test/**/*.json')
             for json_path in json_paths:
                 with open(json_path) as json_file:
-                    json_string = simplejson.loads(json_file.read())
+                    json_string = simplejson.loads(json_file.read(), allow_nan=True)
                 track_df = json2df(json_string, json_path.stem)
                 method.add_track(track_df)
         self.add_evalstore(method, p.stem)

From da27cfa35fcdbd9608667caff7ff9feab5e20bf0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?David=20D=C3=ADaz-Guerra=20Aparicio?= <ddga@unizar.es>
Date: Mon, 22 May 2023 15:35:49 +0300
Subject: [PATCH 3/3] Add minimum version of simplejson

The fix for the new behavior of simplejson.load is not backward compatible.
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index f990a2d..96da919 100644
--- a/setup.py
+++ b/setup.py
@@ -47,7 +47,7 @@
             'pandas>=1.0.1',
             'numpy',
             'scipy',
-            'simplejson',
+            'simplejson>=3.19.0',
             'soundfile',
             'jsonschema'
         ],
